backend/pom.xml

```diff
@@ -124,6 +124,19 @@
             <version>${quarkus-logging-cloudwatch.version}</version>
         </dependency>
 
+        <!-- Kafka -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-reactive-messaging-kafka</artifactId>
```

Comment:
FYI we have a "zero Kafka" policy in `notifications-backend` to make sure our frontend never goes down when there's a Kafka outage (which happened several times in the past). You can work around that by forwarding the REST request by `notifications-backend` to a `notifications-engine` REST API using `quarkus-rest-client-reactive`.

Reply:
I see. Then my idea was to:

* Create a `POST /endpoints/test` which takes an `{"orgId": "whatever"}` as the payload, since we don't have the `x-rh-identity` utils to extract the orgId from the headers. And instead of repeating code, I was thinking on just receiving that payload.
* Create a `POST /endpoints/<uuid>/test` on the backend, which extracts the orgId and makes sure that the endpoint exists, before sending the order to the engine.

What do you think?

Reply:
It makes sense, we don't need to verify the `x-rh-identity` header in the engine. It's ok to trust notifications-backend with the  `endpointId`. `notifications-backend` can verify that the `orgId` of the caller matches the one in the `endpoint`.

Reply:
I've been talking to Gwenneg and we finally may go the Kafka route instead of doing it via REST endpoints on our side. We were thinking on adding the `endpointId` to the `Context` of the `Action`, so that we can pull that ID and just skip all the behaviour group dance. What do you think @josejulio?

Reply:
It makes sense to me - that way we can put this in a queue and let the engine process it normally - We might come up with mechanism to notify the user about the status of the endpoint - it could be maybe a notification in the drawer and I suppose this event should appear in the event log as "manual test" or something?

Reply:
Hey @josejulio,

Sorry about the delay, I forgot about this PR. 

> I suppose this event should appear in the event log as "manual test" or something?

Do I have to set up the event or event type in some particular way to make it show that text? For now I've created a custom event type, but I don't know if that's enough.

Thank you!

Reply:
I'm not enterely sure. 

Right now our event log shows as:

![Screenshot_2022-12-01_12-40-09](https://user-images.githubusercontent.com/3845764/205144586-83ddc690-77ec-43c8-86d7-4b6ad2050da7.png)

If we are going to go with the route of using the `endpointId` in the action, we might not even need a real event - we would only need some way to mark the notification history so that we know it was a manual run.

We will have to modify the UI to detect it was a manual run to show something different (Manual testing or so)

Reply:
Mmmmm... I added a new bundle and an application via a migration, which tie to the event type. Can the UI use that to show it as a test event, maybe?

Reply:
> I've been talking to Gwenneg and we finally may go the Kafka route instead of doing it via REST endpoints on our side.

I don't remember what we said several months ago but unless there is a very strong reason that I forgot, we cannot introduce Kafka in `notifications-backend`.

Reply:
I vaguely recall that we mentioned that on the ticket there were two suggestions to implement, and that the suggestion number 2 was the one we should probably implement. But to be honest, I think I misunderstood what we talked back then. I'm sorry for the confusion, I'll go back to the REST API implementation for the test events.

Reply:
Reworked!